lfs/config.go

```diff
@@ -105,38 +105,47 @@ func (c *Configuration) RemoteEndpoint(remote string) Endpoint {
 	if len(remote) == 0 {
 		remote = defaultRemote
 	}
-
 	if url, ok := c.GitConfig("remote." + remote + ".lfsurl"); ok {
-		return Endpoint{Url: url}
+		return NewEndpoint(url)
 	}
 
 	if url, ok := c.GitConfig("remote." + remote + ".url"); ok {
-		return remoteEndpointFromUrl(url)
+		return NewEndpointFromCloneURL(url)
 	}
 
 	return Endpoint{}
 }
 
-func remoteEndpointFromUrl(url string) Endpoint {
+const ENDPOINT_URL_UNKNOWN = "<unknown>"
+
+// Create a new endpoint from a URL associated with a git clone URL
+// The difference to NewEndpoint is that it appends [.git]/info/lfs to the URL since it
+// is the clone URL
```

Comment:
In the future I think method comments should follow [standard godoc conventions](http://blog.golang.org/godoc-documenting-go-code).

Obviously this project isn't strict about docs. But I think if someone takes the time to write something, it should follow the conventions. It's easy enough that I may just make the change live before merging this, or applying it in a quick PR.